python/setup.py

```diff
@@ -195,7 +195,20 @@ def finalize_options(self):
                       extra_compile_args=['-std=c++17'])
         ]
 
-        self.distribution.ext_modules = extensions
+        # Handle nthreads separately to mimic what Cython does
+        nthreads = getattr(self, 'parallel', None)  # -j option in Py3.5+
+        nthreads = int(nthreads) if nthreads else None
+
+        # Delay import this to allow for Cython-less installs
+        from Cython.Build import cythonize
+
+        self.distribution.ext_modules = cythonize(
+            extensions,
+            nthreads=nthreads,
+            compiler_directives=dict(
+                profile=False, language_level=3, embedsignature=True
+            ),
+        )
```

Comment:
Following offline chat, just checked locally, and these changes are not needed, they are re-creating the functionality in https://github.com/rapidsai/cuml/blob/cae8034019c5448fe28f6ffdd1fc0b84c966fe94/python/cython_build_ext.py#L37 (that also includes functionality for the links to source in the docs)

Reply:
(Would be great to uniformize all of our Cython build codes, and retake the work done for Scikit build in #3367 )